dms/models/directory.py

```diff
@@ -272,6 +272,71 @@ def check_access(self, operation, raise_exception=False):
             res = True
         return res
 
+    allowed_model_ids = fields.Many2many(
+        compute="_compute_allowed_model_ids", comodel_name="ir.model", store=False
+    )
+    model_id = fields.Many2one(
+        comodel_name="ir.model",
+        domain="[('id', 'in', allowed_model_ids)]",
+        compute="_compute_model_id",
+        inverse="_inverse_model_id",
+        string="Model",
+        store=False,
+    )
+
+    res_model = fields.Char(string="Res model")
```

Comment:
Better to put a proper label

```suggestion
    res_model = fields.Char(string="Linked attachments model")
```